src/pages/resultsView/coExpression/CoExpressionPlotUtils.ts

```diff
@@ -0,0 +1,117 @@
+import _ from "lodash";
+
+export function logScale(nonNegVal:number) {
+    return Math.log(nonNegVal+1);// +1 to make logs positive
+}
+
+export function invLogScale(val:number) {
+    // needs to be inverse of logScale
+    return Math.exp(val)-1;
+}
+
+export function computePlotData<T extends {x:number, y:number}>(
+    sourceData: T[],
+    paddingFraction:number,
+    xLogScale:boolean,
+    yLogScale:boolean
+):{
+    max: {x:number, y:number},
+    data: (T & {trueX:number, trueY:number})[],
+    trueMin:{x:number, y:number},
+    trueMax:{x:number, y:number},
+    xPadding:number,
+    yPadding:number,
+    xLogScale:boolean,
+    yLogScale:boolean
```

Comment:
Maybe define a type/interface for this, something like `CoExpressionPlotData`? I guess we also pass this same plot data to the `chartSpaceToDataSpace` function.

Reply:
yeah good idea